Build failure: curaengine

### Steps To Reproduce

Steps to reproduce the behavior:

1. build *curaengine*

### Build log

```
error: builder for '/nix/store/jrqs5n1jj6dqd7spfa7xhks43wwiaiib-curaengine-4.13.1.drv' failed with exit code 2;
       last 10 log lines:
       >    22 |     int img_x, img_y, img_z; // stbi requires pointer to int rather than to coord_t
       >       |                ^~~~~
       > /build/source/src/infill/ImageBasedDensityProvider.cpp:24:18: warning: 'img_x' may be used uninitialized [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wmaybe-uninitialized-Wmaybe-uninitialized8;;]
       >    24 |     image_size = Point3(img_x, img_y, img_z);
       >       |                  ^~~~~~~~~~~~~~~~~~~~~~~~~~~
       > /build/source/src/infill/ImageBasedDensityProvider.cpp:22:9: note: 'img_x' declared here
       >    22 |     int img_x, img_y, img_z; // stbi requires pointer to int rather than to coord_t
       >       |         ^~~~~
       > make[1]: *** [CMakeFiles/Makefile2:113: CMakeFiles/_CuraEngine.dir/all] Error 2
       > make: *** [Makefile:156: all] Error 2
       For full logs, run 'nix log /nix/store/jrqs5n1jj6dqd7spfa7xhks43wwiaiib-curaengine-4.13.1.drv'.
```

### Additional context

This was on an update yesterday to `nixpkgs-unstable`.

### Notify maintainers

@abbradar @gebner

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.6.8-zen1, NixOS, 24.05 (Uakari), 24.05.20240114.dd5621d`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.1`
 - channels(kiara): `"home-manager"`
 - channels(root): `""`
 - nixpkgs: `/nix/store/mkj2la46j2lp83rx0mgwz2awi8l6m1wp-source`
```

--- a/pkgs/applications/misc/curaengine/default.nix
+++ b/pkgs/applications/misc/curaengine/default.nix
@@ -1,4 +1,4 @@
-{ lib, stdenv, fetchFromGitHub, cmake, libarcus, stb, protobuf }:
+{ lib, stdenv, fetchFromGitHub, cmake, libarcus, stb, protobuf, fetchpatch }:
 
 stdenv.mkDerivation rec {
   pname = "curaengine";
@@ -15,6 +15,15 @@ stdenv.mkDerivation rec {
   buildInputs = [ libarcus stb protobuf ];
 
   cmakeFlags = [ "-DCURA_ENGINE_VERSION=${version}" ];
+
+  # TODO already fixed in master, remove in next release
+  patches = [
+    (fetchpatch {
+      url = "https://github.com/Ultimaker/CuraEngine/commit/de60e86a6ea11cb7d121471b5dd192e5deac0f3d.patch";
+      hash = "sha256-/gT9yErIDDYAXvZ6vX5TGlwljy31K563+sqkm1UGljQ=";
+      includes = [ "src/utils/math.h" ];
+    })
+  ];
 
   meta = with lib; {
     description = "A powerful, fast and robust engine for processing 3D models into 3D printing instruction";

